02-basics-2/50-selected-meetup/SelectedMeetupApp.js

```diff
@@ -1,75 +1,64 @@
-import { defineComponent } from 'vue'
-// import { getMeetup } from './meetupsService.ts'
+import { defineComponent, ref, watch } from 'vue'
+import { getMeetup } from './meetupsService.ts'
 
 export default defineComponent({
   name: 'SelectedMeetupApp',
 
-  setup() {},
+  setup() {
+    const numbers = ref([1, 2, 3, 4, 5])
+    const title = ref('')
+    const selectedId = ref(numbers.value[0])
+
+    watch(async () => {
+      if (selectedId.value) {
+        const data = await getMeetup(selectedId.value)
+        title.value = data.title
+      }
+    })
```

Comment:
It works, but it is not a valid `watch` usage.

The first argument is `target` - the reactive variable that we watch. The second variable - a handler, a function which will be called when target changes.

A target can be a variable (e.g. `selectedId`) or a getter function, when we need to compute a value (e.g. `() => numbers.value[0]` if we want to watch the first array item).

Your code works, because it uses an entire function as a getter and calls it each time it's dependencies changes. This is exactly what `watchEffect` does.

So, here you should either use `watchEffect` or use `watch` with explicitly set target and getter.

Reply:
I think I solved it :D